src/app/static/src/app/store/plotSelections/utils.ts

```diff
@@ -0,0 +1,145 @@
+import { Commit } from "vuex";
+import { CalibrateMetadataResponse, FilterOption, FilterRef, FilterTypes, PlotSettingOption } from "../../generated";
+import { PlotName } from "./plotSelections";
+import { PlotSelectionUpdate, PlotSelectionsMutations } from "./mutations";
+import { RootState } from "../../root";
+
+export const filtersAfterUseShapeRegions = (filterTypes: FilterTypes[], rootState: RootState) => {
+    const filters = [...filterTypes];
+    const area = filters.find(f => f.id == "area");
+    if (area && area.use_shape_regions) {
+        const regions: FilterOption[] = rootState.baseline.shape!.filters!.regions ?
+            [rootState.baseline.shape!.filters!.regions] : [];
+
+        const index = filters.findIndex(f => f.id === "area");
+        const { use_shape_regions, ...areaFilterConfig } = area;
+        filters[index] = { ...areaFilterConfig, options: regions };
+    }
+    return filters;
+};
+
+type NestedFilterOption = {
+    id: string
+    label: string,
+    children?: NestedFilterOption[]
+}
+
+class Q {
+    items: NestedFilterOption[]
+    constructor() { this.items = [] }
+    enqueue(item: NestedFilterOption) { this.items.push(item) }
+    dequeue() { return this.items.pop()! }
+    isEmpty() { return this.items.length === 0 }
+}
+
+const getFullNestedFilters = (filterOptions: NestedFilterOption[]) => {
+    const fullFilterOptions: NestedFilterOption[] = [];
+    const q = new Q();
+    filterOptions.forEach(op => q.enqueue(op));
+    while (!q.isEmpty()) {
+        const currentOption = q.dequeue();
+        fullFilterOptions.push(currentOption);
+        if (currentOption.children && currentOption.children.length > 0) {
+            currentOption.children.forEach(op => q.enqueue(op));
+        }
+    }
+    return fullFilterOptions;
+};
```

Comment:
Does the queue add much here? Might be simpler just to use the functions from the array instead of adding the extra class? 

But also, do we have this logic duplicated in the tree select? Is there a way we can use the logic from the treeselect here instead of repeating it?

Reply:
the queue class doesnt add much, can just do an array, i can potentially just expose this function from the tree select but i wonder if we shouldnt integrate the package and out state setup that tightly, this one might potentially change in the future or the treeselect one might change